### PR TITLE
Fix non-deterministic ordering in interceptor code generation

### PR DESCRIPTION
**Pull request by h0tk3y**

### CODE REVIEW

**Comment by h0tk3y** [comment]
@bot-gradle test RFN

File: subprojects/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/AbstractInstrumentationProcessor.java [hunk 1]
```diff
@@ -104,6 +105,8 @@ private void collectAndProcessRequests(Collection<? extends Element> annotatedEl
     @Nonnull
     private static List<ExecutableElement> getExecutableElementsFromAnnotatedElements(Collection<? extends Element> annotatedClassElements) {
         return annotatedClassElements.stream()
+            // Ensure that the type elements have a stable order, as the annotation processing engine does not guarantee it
+            .sorted(Comparator.comparing(AbstractInstrumentationProcessor::elementQualifiedNameIfPresent))
```
Comment (bamboo) [inline subprojects/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/AbstractInstrumentationProcessor.java:109]: ❓ Wouldn't it be better to sort it after it has been filtered? Or even better, after all elements are known to `ExecuteElement` instances only?

File: subprojects/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/codegen/CompositeInstrumentationCodeGenerator.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ public GenerationResult generateCodeForRequestedInterceptors(Collection<CallInte
         }
 
         List<CanGenerateClasses> generatingResults = results.stream().map(it -> (CanGenerateClasses) it).collect(Collectors.toList());
-        Map<String, CanGenerateClasses> generatorByClassName = new HashMap<>();
+        Map<String, CanGenerateClasses> generatorByClassName = new LinkedHashMap<>();
```
Comment (bamboo) [inline subprojects/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/codegen/CompositeInstrumentationCodeGenerator.java:49]: ❓ I assume the order of the `generatingResults` collection is stable as well?

Comment (h0tk3y) [reply subprojects/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/codegen/CompositeInstrumentationCodeGenerator.java:49]: Yes, it is, as long as all the generator implementations produce the results in a stable order.

**Comment by bamboo** [comment]
@bot-gradle test this

**Comment by bamboo** [comment]
@bot-gradle test and merge

File: subprojects/internal-instrumentation-processor/src/test/groovy/org/gradle/internal/instrumentation/processor/ConfigurationCacheInstrumentationProcessorTest.groovy [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                public static void ${second}() {}
+            }
+        """
+    }
+
+    private void sourceFile(String name, @Language('java') String source) {
+        srcDir.file(name).text = source.stripIndent()
+    }
+
+    private DiagnosticCollector<JavaFileObject> processAnnotationsOf(Iterable<String> classes) {
```
Comment (h0tk3y) [inline subprojects/internal-instrumentation-processor/src/test/groovy/org/gradle/internal/instrumentation/processor/ConfigurationCacheInstrumentationProcessorTest.groovy:133]: 👍 Wow, I did not expect the setup for the test to be so compact!

Comment (bamboo) [reply subprojects/internal-instrumentation-processor/src/test/groovy/org/gradle/internal/instrumentation/processor/ConfigurationCacheInstrumentationProcessorTest.groovy:133]: ChatGPT did it 😉

**Pull request merged**